File: src/c1removeadjacentduplicates/Java/RemoveDuplicatesHelper.java
package c1removeadjacentduplicates.Java;

import java.util.Stack;

public class RemoveDuplicatesHelper {
    public static String removeDuplicates(String s) {
        Stack<Character> stack = new Stack<>();

        for (char c : s.toCharArray()) {
            if (!stack.isEmpty() && stack.peek() == c) {
                stack.pop(); // pop the last element
            } else {
                stack.push(c); // push the current character
            }
        }

        // Convert stack to string
        StringBuilder result = new StringBuilder();
        while (!stack.isEmpty()) {
            result.insert(0, stack.pop()); // Stack needs to be reversed to get correct order
        }
        return result.toString();
    }
}

File: src/c1removeadjacentduplicates/CS/StacksRAD.Algo/RemoveDuplicatesHelper.cs
// File: RemoveDuplicatesHelper.cs

namespace StacksRAD.Algo;

public static class RemoveDuplicatesHelper
{
    public static string RemoveDuplicates(string s)
    {
        Stack<char> stack = new();

        foreach (char c in s)
        {
            if (stack.Count > 0 && stack.Peek() == c)
            {
                stack.Pop(); // pop the last element
            }
            else
            {
                stack.Push(c); // push the current character
            }
        }

        // Convert stack to string
        char[]? resultArray = [.. stack];
        Array.Reverse(resultArray); // Stack needs to be reversed to get correct order

        return new string(resultArray);
    }
}
